app/menus/members/menu.rb

```diff
@@ -24,50 +24,49 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 #
 # See COPYRIGHT and LICENSE files for more details.
-#++
+# ++
+module Members
+  class Menu < Submenu
+    attr_reader :project, :params
 
-module Menus
-  module MembersHelper
-    def first_level_menu_items
-      [OpenProject::Menu::MenuGroup.new(header: nil, children: user_status_options)] + nested_menu_items
-    end
-
-    private
+    def initialize(project: nil, params: nil)
+      @project = project
+      @params = params
 
-    def user_status_options
-      [
-        OpenProject::Menu::MenuItem.new(title: I18n.t("members.menu.all"),
-                                        href: project_members_path,
-                                        selected: active_filter_count == 0),
-        OpenProject::Menu::MenuItem.new(title: I18n.t("members.menu.locked"),
-                                        href: project_members_path(status: :locked),
-                                        selected: selected?(:status, :locked)),
-        OpenProject::Menu::MenuItem.new(title: I18n.t("members.menu.invited"),
-                                        href: project_members_path(status: :invited),
-                                        selected: selected?(:status, :invited))
-      ]
+      super(view_type:, project:, params:)
```

Comment:
`view_type` will be `nil` as it resolves to `@view_type` from the parent class `attr_reader :view_type`. It's not very obvious. I'd make it explicitly `nil` instead.

Was it intended?

Reply:
I set it explcitly to `nil` in a commit, but I still wonder if it's correct to do so. (I have not understood what this `view_type` is used for).

Reply:
Yes, it is intended to be `nil`. The `view_type` is a reference for each query that somehow deals with WokPackages in which module it should be shown. However, for members this is not relevant